mesonbuild/interpreter/mesonmain.py

```diff
@@ -347,6 +348,16 @@ def override_dependency_method(self, args: T.Tuple[str, dependencies.Dependency]
         if not name:
             raise InterpreterException('First argument must be a string and cannot be empty')
 
+        # Make a copy since we're giong to mutate. This avoids situations like:
+        #
+        #   dep = declare_dependency()
+        #   meson.override_dependency('foo', dep)
+        #   meson.override_dependency('foo-1.0', dep)
+        #   dep = dependency('foo')
+        #   dep.name() # == 'foo-1.0'
```

Comment:
Can you turn that into a unit test please?

Reply:
good call. Done